### PR TITLE
Refactor App

### PR DESCRIPTION
**Pull request by sayatul**
[FP-5157]
* Python 3 Support
* Added Gitlint
* Updated requirements
* Added CricleCI boiler plate

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -42,7 +41,7 @@ def clean(ctx):
 
 
 def _run_flask_command(ctx, command):
-    ctx.run('FLASK_APP={0} flask {1}'.format(AUTOAPP, command), echo=True)
+    ctx.run('FLASK_APP={0} flask {1}'.format(MANAGE, command), echo=True)
```
Comment (Anmol-Gulati) [inline tasks.py]: What is this whole file suppose to do?

File: {{cookiecutter.app_name}}/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+FROM python:3.6
 
 # copy the rest of the app
 COPY . /app
 
 # Workdir
 WORKDIR /app
-RUN pip install -r requirements.txt
-CMD "gunicorn autoapp:app"
+RUN pip install -r requirements/prod.txt
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/Dockerfile]: Previous one was fine, I guess

File: {{cookiecutter.app_name}}/README.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ In general, before running shell commands, set the ``FLASK_APP`` and
 ``FLASK_DEBUG`` environment variables:
 
 ```sh
-    export FLASK_APP=/path/to/autoapp.py
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/README.md:44]: Can you make a separate heading for ENV Variables like this https://github.com/fulfilio/bifrost/blob/develop/README.md ?

File: {{cookiecutter.app_name}}/migrations/env.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                logger.info('No changes in schema detected.')
+
+    engine = engine_from_config(config.get_section(config.config_ini_section),
+                                prefix='sqlalchemy.',
+                                poolclass=pool.NullPool)
+
+    connection = engine.connect()
+    context.configure(connection=connection,
+                      target_metadata=target_metadata,
+                      process_revision_directives=process_revision_directives,
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/migrations/env.py:75]: https://github.com/fulfilio/fulfil-3PL/blob/develop/migrations/env.py#L76 Add this also

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/extensions.py [hunk 5]
```diff
@@ -18,55 +17,17 @@
...
-
-
 # Database
 db = SQLAlchemy()
 migrate = Migrate()
 
 
 def get_fulfil():
-    # Check if set on global. This happens from
-    # celery tasks
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/extensions.py:69]: Why just remove the comments? Remove the related code also

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/settings.py [hunk 6]
```diff
@@ -10,9 +10,12 @@ class Config(object):
 
     if not get_debug_flag():
         # Assert env for prod
-        for key in ('ENCR_KEY', '{{ cookiecutter.app_name|upper }}_SECRET_KEY', 'DATABASE_URI', 'CELERY_BROKER_URL'):
+        keys = (
+            'ENCR_KEY', '{{ cookiecutter.app_name|upper }}_SECRET_KEY',
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/settings.py:14]: DATABASE_URI ?

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/models.py [hunk 7]
```diff
@@ -9,48 +9,52 @@
...
+        if not merchant:
+            merchant = cls()
+        merchant.name = name
+        merchant.subdomain = subdomain
+        merchant.token = token
+        merchant.save()
+        return merchant
 
 
 class User:
```
Comment (Anmol-Gulati) [inline {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/models.py:60]: Where is this class being used?

**Review by Anmol-Gulati — CHANGES_REQUESTED** [review]

File: {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/models.py (hunk 7 above)
Comment (sayatul) [reply {{cookiecutter.app_name}}/{{cookiecutter.app_name}}/user/models.py:60]: in extensions.py https://github.com/fulfilio/cookiecutter-fulfil/pull/1/files#diff-950097abae480635b3beb2ea5778ebc5R57

**Comment by Anmol-Gulati** [comment]
Add a Procfile too for heroku, maybe this can be a part of config too. Ask the developer for deployment platform

**Pull request merged**